glibc 2.30 added gettid() which conflicts with the implementation in
NetworkManager.  Remove for 1.17.1 and later versions.

Adapted from upstream:
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/commit/10276322bde8f015e48ac06f6a7509f514eb46f0

--- a/src/systemd/sd-adapt/nm-sd-adapt.h
+++ b/src/systemd/sd-adapt/nm-sd-adapt.h
@@ -181,9 +181,10 @@
 #endif
 }
 
-static inline pid_t gettid(void) {
+static inline pid_t _nm_gettid(void) {
         return (pid_t) syscall(SYS_gettid);
 }
+#define gettid() _nm_gettid ()
 
 /* we build with C11 and thus <uchar.h> provides char32_t,char16_t. */
 #define HAVE_CHAR32_T 1
